src/platform/Darwin/DnssdImpl.h

```diff
@@ -47,13 +50,14 @@ struct RegisterContext : public GenericContext
 
     RegisterContext(const char * sType, DnssdPublishCallback cb, void * cbContext)
     {
-        type = ContextType::Register;
-        strncpy(mType, sType, sizeof(mType));
+        type     = ContextType::Register;
         context  = cbContext;
         callback = cb;
+
+        Platform::CopyString(mType, sType);
     }
 
-    bool matches(const char * sType) { return (strcmp(mType, sType) == 0); }
+    bool matches(const char * sType) { return (::std::strcmp(mType, sType) == 0); }
```

Comment:
Style-wise we usually include <string.h> and avoid std:: prefix for things that come from C rather than STL, such as `str*`. There is a 10:1 usage differential between <string.h> and <cstring.h>